### PR TITLE
feat(parser): Adds support for parsing order by in functions

### LINKED ISSUE #3434: frontend: support parsing agg order
Some aggregation functions contain orders in the SQL, like
```sql
select string_agg(first_name || ' ' || last_name, ', ' order by last_name, first_name) from users
```

We should add the support for the parser first.

Implementation from Postgres: https://github.com/postgres/postgres/blob/7103ebb7aae8ab8076b7e85f335ceb8fe799097c/src/backend/parser/parse_agg.c#L76-L104

### PR DESCRIPTION
**Pull request by Graphcalibur**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?
- Summarize your change (**mandatory**)
    - See PR title.

## Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Refer to a related PR or issue link (optional)
Fixes #3434

### CODE REVIEW

File: src/sqlparser/tests/sqlparser_common.rs [hunk 1]
```diff
@@ -1827,6 +1832,40 @@ fn parse_aggregate_with_group_by() {
     // TODO: assertions
 }
 
+#[test]
+fn parse_aggregare_with_order_by() {
```
Comment (yuhao-su) [inline src/sqlparser/tests/sqlparser_common.rs:1836]:
```suggestion
fn parse_aggregate_with_order_by() {
```

File: src/sqlparser/src/parser.rs [hunk 2]
```diff
@@ -561,7 +561,7 @@ impl Parser {
     pub fn parse_function(&mut self, name: ObjectName) -> Result<Expr, ParserError> {
         self.expect_token(&Token::LParen)?;
         let distinct = self.parse_all_or_distinct()?;
-        let args = self.parse_optional_args()?;
+        let (args, order_by) = self.parse_optional_args()?;
```
Comment (BugenZhao) [inline src/sqlparser/src/parser.rs:564]: Do we need to check the function name before deciding whether `ORDER BY` is acceptable?

Comment (Graphcalibur) [reply src/sqlparser/src/parser.rs:564]: I think it is ok to accept `ORDER BY` without checking the function name since Postgresql seems to do the same.

Comment (xiangjinwu) [reply src/sqlparser/src/parser.rs:564]: The check can be done in binder.

File: src/sqlparser/src/parser.rs [hunk 3]
```diff
@@ -2780,10 +2781,10 @@ impl Parser {
         } else {
             let name = self.parse_object_name()?;
             // Postgres,table-valued functions:
-            let args = if self.consume_token(&Token::LParen) {
+            let (args, _) = if self.consume_token(&Token::LParen) {
```
Comment (xiangjinwu) [inline src/sqlparser/src/parser.rs:2784]: Either report an error here when `order_by` is not empty, or return it as part of `TableFactor::Table` and let the upper level decide whether it is valid.

**Review by xiangjinwu — APPROVED** [review]
rest lgtm

**Pull request merged**